### PR TITLE
feat(card): expose global card css variable

### PR DESCRIPTION
**Pull request by liamdebeasi**
Docs PR: https://github.com/ionic-team/ionic-docs/pull/1542
Starters PR: https://github.com/ionic-team/starters/pull/1393

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: resolves https://github.com/ionic-team/ionic-framework/issues/21694

## What is the new behavior?

- The card bg color on dark mode was wrong because it follows `ion-item`'s background color
- I exposed a `--ion-card-background` global variable so we can separate the two and properly set the background color for the card.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by liamdebeasi** [comment]
One thing to consider here -- are there people styling `--ion-item-background` and hoping that `ion-card` follows suit? If so, this may be a breaking change. Perhaps instead of:

```scss
$card-ios-background:                        var(--ion-card-background, $background-color) !default;
```

we could do

```scss
$card-ios-background:                        var(--ion-card-background, $item-ios-background) !default;
```

**Comment by brandyscarney** [comment]
> One thing to consider here: Are there people styling `--ion-item-background` and hoping that `ion-card` follows suit? If so, this may be a breaking change. Perhaps instead of:
> 
> ```css-scss
> $card-ios-background:                        var(--ion-card-background, $background-color) !default;
> ```
> 
> we could do
> 
> ```css-scss
> $card-ios-background:                        var(--ion-card-background, $item-ios-background) !default;
> ```

Yeah agreed that a fallback to item background would be desired.

Should we include a variable for color? I think if they change the card color to something far off from the default text color it would be beneficial.

**Comment by liamdebeasi** [comment]
Do we include a global var for color for ion-item?

**Comment by brandyscarney** [comment]
Yis

```
$item-ios-background:                       var(--ion-item-background, $background-color) !default;
$item-ios-border-color:                     var(--ion-item-border-color, var(--ion-border-color, var(--ion-color-step-250, #c8c7cc))) !default;
$item-ios-color:                            var(--ion-item-color, $text-color) !default;
```

**Comment by liamdebeasi** [comment]
Cool yeah I think that makes sense. I'll make those changes 👍

**Comment by liamdebeasi** [comment]
@brandyscarney Ok changes made. Mind taking another look when you get a chance?

File: core/src/themes/ionic.theme.default.ios.scss [hunk 1]
```diff
@@ -29,4 +29,9 @@ $toolbar-ios-color:                         var(--ion-toolbar-color, $text-color
...
 $item-ios-background:                       var(--ion-item-background, $background-color) !default;
 $item-ios-border-color:                     var(--ion-item-border-color, var(--ion-border-color, var(--ion-color-step-250, #c8c7cc))) !default;
-$item-ios-color:                            var(--ion-item-color, $text-color) !default;
\ No newline at end of file
+$item-ios-color:                            var(--ion-item-color, $text-color) !default;
+
+// iOS Card
+// --------------------------------------------------
+$card-ios-background:                        var(--ion-card-background, $item-ios-background) !default;
+$card-ios-color:                             var(--ion-card-color, $text-color-step-400) !default;
```
Comment (brandyscarney) [inline core/src/themes/ionic.theme.default.ios.scss:37]:
```suggestion
$card-ios-color:                             var(--ion-card-color, var(--ion-item-color, $text-color-step-400)) !default;
```

File: core/src/themes/ionic.theme.default.md.scss [hunk 2]
```diff
@@ -31,3 +31,8 @@ $toolbar-md-color:                          var(--ion-toolbar-color, var(--ion-t
 $item-md-background:                        var(--ion-item-background, $background-color) !default;
 $item-md-border-color:                      var(--ion-item-border-color, var(--ion-border-color, var(--ion-color-step-150, rgba(0, 0, 0, .13)))) !default;
 $item-md-color:                             var(--ion-item-color, $text-color) !default;
+
+// Material Design Card
+// --------------------------------------------------
+$card-md-background:                        var(--ion-card-background, $item-md-background) !default;
+$card-md-color:                             var(--ion-card-color, $text-color-step-450) !default;
```
Comment (brandyscarney) [inline core/src/themes/ionic.theme.default.md.scss:38]:
```suggestion
$card-md-color:                             var(--ion-card-color, var(--ion-item-color, $text-color-step-450)) !default;
```

**Pull request merged**